core/trino-main/src/main/java/io/trino/testing/assertions/Assert.java

```diff
@@ -71,7 +77,7 @@ public static void assertEventually(Duration timeout, Runnable assertion)
                 }
             }
             try {
-                Thread.sleep(50);
+                Thread.sleep(retryDuration.toMillis());
```

Comment:
i wonder what are the cases when 50ms is a wrong value.
Maybe we can derive a better waitBeforeRetry value based on timeout?
Eg. for timeout in minutes, sleeping second or few seconds sounds more reasonable

Reply:
for example, the sleep could be 1/1000 of timeout eliminating the need for a parameter

Reply:
I like that. Not sure about the `1/1000` though. What are timeouts typically?

Reply:
> Maybe we can derive a better waitBeforeRetry value based on timeout?

That can be done independently of this PR